lib/materials/data_helpers.rb

```diff
@@ -424,13 +424,17 @@ def links_for_material( material,
         }
       end
 
+      assignPopupConfig = { :material_id => material.id, :material_type => material.class.to_s, :lightbox_material_text => t('material').pluralize.capitalize, :skip_reload => skip_lightbox_reloads }
       if current_visitor.portal_teacher && material.respond_to?(:offerings)
-        links[:assign_material] = {
-            text: "Assign to a Class",
-            url: "javascript:void(0)",
-            onclick: "get_Assign_To_Class_Popup(#{material.id},'#{material.class.to_s}','#{t('material').pluralize.capitalize}',#{skip_lightbox_reloads})"
-        }
+        assignPopupConfig[:anonymous] = false
+      else
```

Comment:
It would be good to handle the case of a logged in student finding their way to this code somehow. The current conditions will treat them as anonymous